### DISCUSSION TITLE
Disable desktop compositing on linux to reduce latency

### DISCUSSION THREAD

**Discussion by chicco-carone** [ppy/osu#23871] [Ideas]
On linux using desktop enviroments that have a compositor (ex gnome or kde) disabling it makes the game feel way more fluid and runs with a lot less input latency. For me using kde i have a kwin script that disables the compositor when a fullscreen application is launched but disabling and enabling it and comparing the game it makes a lot of difference. I have the same outcome using stable on wine. Some programs like lutris have a toggle to disable desktop effects after launching a game so I think it could be implemented as a toggle on the linux client. After seeing the latest lazer update video where there were implemented fixes to lower latency on mac os I thought that a simple toggle for the compositor would be a great addition, it would make the game feel better and have a lower input latency. I don't think it's too hard to make

**Comment by peppy** [comment]
If it's within our control, this should at most be automatically applied when running fullscreen.If there's any reason it can't be implemented like this and instead needs to be a toggle, then it shoul dbe left up to the user instead.

**Reply by chicco-carone** [reply]
I think it should be a toggle because if you just disable it when the game is running other windows will lose animations and get black borders. It depends on how it can get implemented because my kwin script disables the composition only when the full screen window is getting used and as soon as it gets in the background it will re enable it so there isn't this issue.

**Reply by peppy** [reply]
In that case it will be up to the user.

**Comment by THEGOLDENPRO** [comment]
May I have this script of yours? 
I'm on Fedora 39 with KDE and I'm experiencing a little bit of input delay on osu! Lazer. I'm hoping this script will help.

**Reply by chicco-carone** [reply]
its in the discover store and its in the kwin scripts section and it's called autocomposer. This only works in x11 if you are in wayland make sure to check the box in the compositor settings to disable the compositor when running fullscreen

**Reply by THEGOLDENPRO** [reply]
ah okay, thank you

**Comment by chicco-carone** [comment]
> ah okay, thank you

if you want to reduce it even more if you use pipewire you can set the variable PIPEWIRE_LATENCY="32/42000" before launching the game. 32 is the lowest that pipewire can handle but that value is dependent a lot from your cpu. for example on my desktop with a 11gen i5 i can handle 32 but on my laptop with a 13 gen i5 it can't handle anything lower 64 so try and see how low you can go before your audio starts crackling
